### PR TITLE
Adding OHDSI Deployment 

### PR DESCRIPTION
**Pull request by timmyreilly**
PR for issue #1505 

Adding deployment for OHDSI with Atlas 
Creating new porter deployment 
Adding terraform 

## How is this addressed

- Describe the changes made, and if appropriate, why they are addressed this way
- Note any pending work (with links to the issues that will address them)
- Update documentation

### CODE REVIEW

File: templates/workspace_services/ohdsi/terraform/variables.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+variable "asp_sku_size" {
+  default = "S1"
+}
+
+variable "broadsea_image" {
+  description = "Docker image for OHDSI - Broadsea (Atlas and WebAPI)"
+  default     = "broadsea-webtools"
+}
+
+# TODO: this should come from ACR
```
Comment (ross-p-smith) [inline templates/workspace_services/ohdsi/terraform/variables.tf:46]: Will this be done in this PR or will you raise another issue to finish this?

File: templates/workspace_services/ohdsi/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Azure Provider source and version being used
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=2.97.0"
```
Comment (ross-p-smith) [inline templates/workspace_services/ohdsi/terraform/main.tf:6]: Use 3.5

File: templates/workspace_services/ohdsi/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  name                = "privatelink.file.core.windows.net"
+  resource_group_name = local.core_resource_group_name
+}
+
+data "azurerm_app_service" "api_core" {
+  name                = "api-${var.tre_id}"
+  resource_group_name = "rg-${var.tre_id}"
+}
+
+data "azurerm_app_service_plan" "asp_core" {
```
Comment (ross-p-smith) [inline templates/workspace_services/ohdsi/terraform/main.tf:91]: Use azurerm_service_plan instead

File: templates/workspace_services/ohdsi/terraform/locals.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+locals {
+  short_service_id               = substr(var.tre_resource_id, -4, -1)
+  short_workspace_id             = substr(var.workspace_id, -4, -1)
+  workspace_resource_name_suffix = "${var.tre_id}-ws-${local.short_workspace_id}"
+  service_resource_name_suffix   = "${var.tre_id}-ws-${local.short_workspace_id}-svc-${local.short_service_id}"
+  webapp_name                    = "guacamole-${local.service_resource_name_suffix}"
```
Comment (ross-p-smith) [inline templates/workspace_services/ohdsi/terraform/locals.tf:6]: cut and paste?

File: templates/workspace_services/ohdsi/terraform/debug.tf [hunk 5]
```diff
@@ -0,0 +1,209 @@
+data "azuread_client_config" "current" {}
+data "azurerm_subscription" "primary" {}
```
Comment (ross-p-smith) [inline templates/workspace_services/ohdsi/terraform/debug.tf:2]: Should this be in the PR?

File: templates/workspace_services/ohdsi/terraform/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export TF_LOG=""
```
Comment (ross-p-smith) [inline templates/workspace_services/ohdsi/terraform/deploy.sh:1]: Does this work? This is called from `make terraform-deploy DIR=./templates/workspace_services/ohdsi`

**Comment by timmyreilly** [comment]
image
Rough draft architecture and workspace/user workspace split.

**Comment by timmyreilly** [comment]
Some advantages of bringing OHDSI to AzureTRE
- Preconfigured admin environment and user experience
- Network Topology Similarity 
- Airlock features

**Comment by marrobi** [comment]
Closing as superseded by https://github.com/microsoft/AzureTRE/pull/3552

**Pull request closed without merging**